### DISCUSSION TITLE
How do I set up Nano for NodeJS?

### DISCUSSION THREAD

**Discussion by AlenVelocity** [nanojsx/nano#49] [Help / Questions] [answered]
Hello, I've been trying to use Nano with node, but it's throwing `'document' is not defined` the following error whenever I call the `h` function or use JSX. My question is, am I missing something? Do I have I have to manually import 'nano-jsx/lib/ssr' or is this intended behavior?

My `tsconfig.json` looks like this
```JSON
{
    "compilerOptions": {
        "target": "es6",
        "lib": [
            "ESNext"
        ],                         
        "module": "commonjs",
        "outDir": "./dist",                        
        "resolveJsonModule": true,
        "strict": true,                           
        "esModuleInterop": true,                  
        "skipLibCheck": true,
        "jsx": "react",   
        "jsxFactory": "Nano.h",              
        "forceConsistentCasingInFileNames": true,
        "experimentalDecorators": true  
    },
    "include": [
        "src/**/*"
    ],
    "exclude": [
        "node_modules",
        "tests/**/*"
    ]
}
```

Additional Info
---

What I'm trying to achieve is, to migrate from [`async-jsx-html`](https://www.npmjs.com/package/async-jsx-html) to Nano
https://github.com/AlenSaito1/Aureolin/issues/33

**Comment by yandeu** [comment]
Please take a look at example1:  
https://github.com/nanojsx/example1

**Comment by Shinyaigeek** [comment, marked as answer]
https://github.com/AlenSaito1/Aureolin/compare/migrate-nanojsx

I took a look at your branch for migrating to nano-jsx.

https://github.com/AlenSaito1/Aureolin/commit/72c225a94178ea564287248a7f81a1381f90a9f2

The document object initialization in this commit is done by `initSSR` function on nano-jsx at

https://github.com/nanojsx/nano/blob/master/src/ssr.ts#L15-L21

This initSSR is executed in the `renderSSR` function. The reason for the error in your application is probably that the `h` function is executed before the `renderSSR` is executed. :eyes:

```typescript
// throw error

const App = h("div", {}, "txt");
renderSSR(App)

// ok
const App = () => h("div", {}, "txt");
renderSSR(<App />);
```

**Reply by AlenVelocity** [reply]
Got it, Thank you.
